src/main/java/modbus/ModbusConnection.java

```diff
@@ -0,0 +1,366 @@
+package modbus;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.dsa.iot.dslink.node.Node;
+import org.dsa.iot.dslink.node.Permission;
+import org.dsa.iot.dslink.node.actions.Action;
+import org.dsa.iot.dslink.node.actions.ActionResult;
+import org.dsa.iot.dslink.node.actions.Parameter;
+import org.dsa.iot.dslink.node.value.Value;
+import org.dsa.iot.dslink.node.value.ValueType;
+import org.dsa.iot.dslink.util.Objects;
+import org.dsa.iot.dslink.util.handler.Handler;
+import org.dsa.iot.dslink.util.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.serotonin.modbus4j.ModbusMaster;
+
+import modbus.SerialConn.AddDeviceHandler;
+
+abstract public class ModbusConnection {
+	private static final Logger LOGGER;
+
+	static {
+		LOGGER = LoggerFactory.getLogger(ModbusConnection.class);
+	}
+
+	static final String ATTR_SLAVE_NAME = "name";
+	static final String ATTR_SLAVE_ID = "slave id";
+	static final String ATTR_POLLING_INTERVAL = "polling interval";
+	static final String ATTR_ZERO_ON_FAILED_POLL = "zero on failed poll";
+	static final String ATTR_USE_BATCH_POLLING = "use batch polling";
+	static final String ATTR_CONTIGUOUS_BATCH_REQUEST_ONLY = "contiguous batch requests only";
+
+	static final String ATTR_TIMEOUT = "Timeout";
+	static final String ATTR_RETRIES = "retries";
+	static final String ATTR_MAX_READ_BIT_COUNT = "max read bit count";
+	static final String ATTR_MAX_READ_READ_REGISYER_COUNT = "max read register count";
+	static final String ATTR_MAX_WRITE_REGISTER_COUNT = "max write register count";
+	static final String ATTR_DISCARD_DATA_DELAY = "discard data delay";
+	static final String ATTR_USE_MULTIPLE_WRITE_COMMAND_ONLY = "use multiple write commands only";
+
+	static final String ATTR_RESTORE_TYPE = "restoreType";
+	static final String ATTR_RESTORE_EDITABLE_FOLDER = "editable folder";
+
+	Node node;
+	Node statnode;
+	ModbusLink link;
+	ModbusMaster master;
+	Set<SlaveNode> slaves;
+	ScheduledFuture<?> reconnectFuture = null;
+	String name;
+	int retryDelay = 1;
+	int timeout;
+	int retries;
+	int maxrbc;
+	int maxrrc;
+	int maxwrc;
+	int ddd;
+	boolean mwo;
+
+	final ScheduledThreadPoolExecutor stpe = Objects.createDaemonThreadPool();
+
+	public ModbusConnection(ModbusLink link, Node node) {
+		this.link = link;
+		this.node = node;
+
+		this.statnode = node.createChild("STATUS").setValueType(ValueType.STRING)
```

Comment:
I think the name shouldn't be capitalized. (Or do you want to reference a constant?)